lib/dal/src/attribute/value.rs

```diff
@@ -2536,36 +2536,67 @@ impl AttributeValue {
         Ok(pairs)
     }
 
-    pub async fn remove_by_id(ctx: &DalContext, id: AttributeValueId) -> AttributeValueResult<()> {
-        let parent_av_id = Self::parent_id(ctx, id)
-            .await?
-            .ok_or(AttributeValueError::RemovingWhenNotChildOrMapOrArray(id))?;
+    /// Remove an array or map element.
```

Comment:
This will also remove object children... so the comment is not quite true

Reply:
A github comment on my review disappeared for some reason (eventual consistency? i guess!).

I wanted to say: why have this error at all. It only prevents deleting the root attribute value. I say remove the check. We should validate the input higher up than this call if we want to prevent the removal of non array or map elements (and this check does not prevent that!)